Common/util/filestream.cpp

```diff
@@ -12,24 +12,23 @@
 //
 //=============================================================================
 
-#if defined(WINDOWS_VERSION)
-#include <io.h>
-#endif
 #include <stdio.h>
 #include "util/filestream.h"
 #include "util/math.h"
 
-// TODO: use fstat on Windows too?
-#if !defined (WINDOWS_VERSION)
-#include <sys/stat.h>
-long int filelength(int fhandle)
-{
-    struct stat statbuf;
-    fstat(fhandle, &statbuf);
-    return statbuf.st_size;
-}
+#if defined(HAVE_FSEEKO) // Contemporary POSIX libc
+    #define file_off_t  off_t
```

Comment:
does your build system support checking things ?
if so, the best thing would be to check whether the off_t type is defined
like by trying to compile this here
```c
#include <unistd.h>
int main() { off_t foo; }
```

Reply:
> does your build system support checking things ?

You mean like CMake? frankly that's the only thing I know that can do that.
No, we don't have anything like that set up right now. Maybe we should do at some point.

Reply:
hell no, [cmake sucks](https://github.com/libtom/libtomcrypt/issues/391#issuecomment-390434739). autoconf could do it, or even plain Makefiles with some tricks.

Reply:
Now implemented in current cmake build system.

Reply:
nice try trolling me